src/Dapr.Extensions.Configuration/DaprSecretDescriptor.cs

```diff
@@ -29,22 +29,46 @@ public class DaprSecretDescriptor
         /// A collection of metadata key-value pairs that will be provided to the secret store. The valid metadata keys and values are determined by the type of secret store used.
         /// </summary>
         public IReadOnlyDictionary<string, string> Metadata { get; }
+        
+        /// <summary>
+        /// This flag indicates if this field's existence should trigger an exception. Setting it to "false"
+        /// will suppress the exception whereas setting it to "true" will not suppress it.
+        /// </summary>
+        public bool IsRequired { get; }
+
+        /// <summary>
+        /// SecretKey value is mapping value to Vault Name. If The application's Secret Name and Secret in the
+        /// Vault Name is different then you can use this flag to specify Vault Secret Name. 
+        /// </summary>
+        public string SecretKey { get; }
 
         /// <summary>
-        /// Secret Descriptor Construcutor
+        /// Secret Descriptor Constructor
         /// </summary>
         public DaprSecretDescriptor(string secretName) : this(secretName, new Dictionary<string, string>())
         {
 
         }
 
         /// <summary>
-        /// Secret Descriptor Construcutor
+        /// Secret Descriptor Constructor
+        /// </summary>
+        public DaprSecretDescriptor(string secretName, IReadOnlyDictionary<string, string> metadata) : 
+            this(secretName, metadata, true, secretName)
+        {
+            
+        }
+        
+        /// <summary>
+        /// Secret Descriptor Constructor
         /// </summary>
-        public DaprSecretDescriptor(string secretName, IReadOnlyDictionary<string, string> metadata)
+        public DaprSecretDescriptor(string secretName, IReadOnlyDictionary<string, string> metadata, 
+            bool isRequired, string secretKey)
```

Comment:
From a style standpoint, if you have to split the parameters, it's best to put each parameter on its own line instead.

Reply:
Fixed